internal/service/advancedclustertpf/update_logic.go

```diff
@@ -0,0 +1,81 @@
+package advancedclustertpf
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/update"
+	"go.mongodb.org/atlas-sdk/v20241113003/admin"
+)
+
+// This function ensures len(state.ReplicationSpecs) == len(plan.ReplicationSpecs) & each element in both state and plan is A) likely the same or B) use an empty replication spec
```

Comment:
```suggestion
// AlignStateReplicationSpecs ensures len(state.ReplicationSpecs) == len(plan.ReplicationSpecs) & each element in both state and plan is A) likely the same or B) use an empty replication spec
```

Go documentation best practice is to start with func name

Reply:
Thanks. Fixed in 7fcd41ef79d5ef4a5a3bee6bf3dfae2895d71134